platform/mbed_error.c

```diff
@@ -152,6 +191,44 @@ static mbed_error_status_t handle_error(mbed_error_status_t error_status, unsign
     return MBED_SUCCESS;
 }
 
+WEAK void mbed_error_reboot_callback(mbed_error_ctx *error_context) {
+    //Dont do anything here, let application override this if required.
+}
+
+//Initialize Error handling system and report any errors detected on rebooted
+mbed_error_status_t mbed_error_initialize(void)
+{
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+    uint32_t crc_val = 0;
+    crc_val = compute_crc32( (unsigned char *)report_error_ctx, ((uint32_t)&(report_error_ctx->crc_error_ctx) - (uint32_t)report_error_ctx) );
```

Comment:
The complicated bit here is  `offsetof(mbed_error_ctx, crc_error_ctx)`, no?

Reply:
Ah yes, thanks for pointing that out and offsetof is much easier for readability.